### PR TITLE
Add docker-compose create command.

### LINKED ISSUE #1125: docker-compose create
Possibility to just create but not run containers. This would be a useful feature for systemd based startups.

### PR DESCRIPTION
**Pull request by seguins**
Adds a new command  for creating a service without starting it.

Closes #1125

### CODE REVIEW

**Comment by aanand** [comment]
LGTM

File: compose/project.py [hunk 1]
```diff
@@ -215,6 +221,22 @@ def build(self, service_names=None, no_cache=False, pull=False, force_rm=False):
...
+
+        plans = self._get_convergence_plans(services, strategy)
+
+        for service in services:
+            plan = plans[service.name]
+            if plan.action == 'create':
+                service.create_container(do_build=True)
+            elif plan.action == 'recreate':
+                service.recreate_container(
+                    service.containers(stopped=True)[0],
```
Comment (dnephin) [inline compose/project.py]: I don't think this is right, there could be more than one container at some point.

File: compose/project.py [hunk 2]
```diff
@@ -215,6 +221,22 @@ def build(self, service_names=None, no_cache=False, pull=False, force_rm=False):
...
+                    do_build=do_build,
+                    start_new_container=False
+                )
```
Comment (dnephin) [inline compose/project.py]: I think it would be good to keep this logic in `service.py`, since it's very similar to the existing `execute_convergence_plan`. Maybe `service.execute_create_convergence_plan()` ? It's a bit wordy, but at least it's clear.

File: tests/acceptance/cli_test.py [hunk 3]
```diff
@@ -264,6 +264,74 @@ def test_build_failed_forcerm(self):
...
+        self.base_dir = 'tests/fixtures/links-composefile'
+        self.dispatch(['create', 'web'], None)
+        web = self.project.get_service('web')
+        db = self.project.get_service('db')
+        console = self.project.get_service('console')
+        self.assertEqual(len(web.containers()), 0)
+        self.assertEqual(len(db.containers()), 0)
+        self.assertEqual(len(web.containers(stopped=True)), 1)
+        self.assertEqual(len(db.containers(stopped=True)), 1)
+        self.assertEqual(len(console.containers(stopped=True)), 0)
```
Comment (dnephin) [inline tests/acceptance/cli_test.py]: Following the [testing pyramid](http://martinfowler.com/bliki/TestPyramid.html) I think the above two tests (create_twice and with_link) should be integration tests in `tests/integration/`. We already have plenty of end-to-end coverage with the other tests you've added.

File: compose/project.py (hunk 2 above)
Comment (aanand) [inline compose/project.py]: `service. execute_convergence_plan(start=False)`?

Comment (seguins) [inline compose/project.py]: @dnephin Thanks for the feedback, I will apply the solution of @aanand which seems cleaner.

File: compose/project.py (hunk 1 above)
Comment (seguins) [inline compose/project.py]: Indeed, it will be removed and fixed by the call to ```service.execute_convergence_plan```

File: compose/project.py (hunk 2 above)
Comment (dnephin) [inline compose/project.py]: I'm a little concerned that is going to make `execute_convergence_plan()` a little overly complex.  We have a max-complexity set to 12, so I guess as long as we don't have to relax that I'm cool with it.

**Comment by seguins** [comment]
@dnephin thanks for the review.
I updated the commit : 
* ```project.create``` uses now ```service.execute_convergence_plan```
* I removed two acceptance tests and add new integration tests

File: compose/project.py [hunk 4]
```diff
@@ -215,6 +221,14 @@ def build(self, service_names=None, no_cache=False, pull=False, force_rm=False):
             else:
                 log.info('%s uses an image, skipping' % service.name)
 
+    def create(self, service_names=None, strategy=ConvergenceStrategy.changed, do_build=True):
+        services = self.get_services_without_duplicate(service_names, include_deps=True)
+
+        plans = self._get_convergence_plans(services, strategy)
+
+        for service in services:
+            service.execute_convergence_plan(plans[service.name], do_build, start=False)
```
Comment (dnephin) [inline compose/project.py]: Just one more small thing, I think we want to avoid calling attach here (it's an unnecessary API call), so this should add `detached=True`  as well

**Comment by dnephin** [comment]
Looks good, just one more small thing

**Comment by seguins** [comment]
Done

**Comment by dnephin** [comment]
LGTM

**Comment by thaJeztah** [comment]
@seguins looks like this needs a rebase :cry:

**Comment by thaJeztah** [comment]
Oh, maybe not, it doesn't say "merge conflict", it's probably github being over restrictive (sorry for the noise :smile:)
